tokio/src/fs/blocking.rs

```diff
@@ -202,9 +203,9 @@ impl Buf {
         self.buf.len() - self.pos
     }
 
-    pub(crate) fn copy_to(&mut self, dst: &mut [u8]) -> usize {
-        let n = cmp::min(self.len(), dst.len());
-        dst[..n].copy_from_slice(&self.bytes()[..n]);
+    pub(crate) fn copy_to(&mut self, dst: &mut dyn BufMut) -> usize {
+        let n = cmp::min(self.len(), dst.remaining_mut());
```

Comment:
we could use `bytes_mut()` slice?

Consider it here and in code below where it can lower number of dynamic method dispatches